src/sentry/snuba/metrics/datasource.py

```diff
@@ -320,7 +330,15 @@ def _fetch_tags_or_values_per_ids(
     # entity by validating that the ids of the constituent metrics all lie in the same entity
     supported_metric_ids_in_entities = {}
 
-    for metric_type in ("counter", "set", "distribution"):
+    release_health_metric_types = ("counter", "set", "distribution")
+    performance_metric_types = ("generic_set", "generic_distribution")
+
+    if use_case_id == UseCaseKey.RELEASE_HEALTH:
+        metric_types = release_health_metric_types
+    else:
+        metric_types = release_health_metric_types + performance_metric_types
```

Comment:
I am not sure I understand why we merge the types if the use case is != release health

Reply:
My bad! Fixed.